bing_search.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# !/usr/bin/env ruby
+
+require 'rest-client'
+
+# A simple Http Request using Rest-Client Bing search for ruby
+
+class BingSearch
+  attr_accessor :url, :querry
+
+  def initialize(url, *querry)
+    @url = url
+    @querry = querry || ' '
+  end
+
+  def response_code
+    rest_client.code
+  end
+
+  def cookies
+    rest_client.cookies
+  end
+
+  def header
+    rest_client.headers
+  end
+
+  def body
+    rest_client.body
+  end
```

Comment:
These should be called on responses. You don't need to make another network call just because you want to check the response code.

One way to do this is to memoize the response.

```
def rest_client
  @response ||= RestClient.get @url, params: { q: @query }
end
```
It reads, `if @response is nil, get data and save it to @response`. So when next you call rest_client, it just returns the @response value.

The other way it to call `#rest_client` once and save the response to @response. Then something like `#body` would be:

```
def body
  @response.body
end
``